### ISSUE TITLE
Serialization/Deserialization segfault

### ISSUE THREAD

**Issue by lstagner** [JuliaLang/julia#32141]
I have been running into this annoying random segfault in my parallel code. I haven't been able to come up with a good MWE as it can't be reliably triggered.
```
julia> versioninfo()
Julia Version 1.1.1
Commit 55e36cc (2019-05-16 04:10 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin15.6.0)
  CPU: Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.1 (ORCJIT, skylake)
Environment:
  JULIA_NUM_THREADS = 12
  JULIA_EDITOR = nvim
```
Here is the error I get.
```
      From worker 8:	
      From worker 8:	signal (11): Segmentation fault: 11
      From worker 8:	in expression starting at no file:0
      From worker 8:	jl_assign_bits at /Users/julia/buildbot/worker/package_macos64/build/src/datatype.c:607
      From worker 8:	jl_set_nth_field at /Users/julia/buildbot/worker/package_macos64/build/src/datatype.c:935
      From worker 8:	jl_new_struct at /Users/julia/buildbot/worker/package_macos64/build/src/datatype.c:794
      From worker 8:	deserialize at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.1/Serialization/src/Serialization.jl:1205
      From worker 8:	handle_deserialize at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.1/Serialization/src/Serialization.jl:782
      From worker 8:	deserialize at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.1/Serialization/src/Serialization.jl:731
      From worker 8:	handle_deserialize at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.1/Serialization/src/Serialization.jl:782
      From worker 8:	deserialize at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.1/Serialization/src/Serialization.jl:731
      From worker 8:	#3 at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.1/Serialization/src/Serialization.jl:869
      From worker 8:	ntuple at ./tuple.jl:136
      From worker 8:	deserialize_tuple at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.1/Serialization/src/Serialization.jl:869
      From worker 8:	handle_deserialize at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.1/Serialization/src/Serialization.jl:761
      From worker 8:	deserialize at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.1/Serialization/src/Serialization.jl:1221
      From worker 8:	handle_deserialize at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.1/Serialization/src/Serialization.jl:782
      From worker 8:	deserialize_fillarray! at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.1/Serialization/src/Serialization.jl:1019
      From worker 8:	deserialize_array at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.1/Serialization/src/Serialization.jl:1011
      From worker 8:	handle_deserialize at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.1/Serialization/src/Serialization.jl:769
      From worker 8:	deserialize at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.1/Serialization/src/Serialization.jl:1221
      From worker 8:	handle_deserialize at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.1/Serialization/src/Serialization.jl:782
      From worker 8:	deserialize at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.1/Serialization/src/Serialization.jl:1221
      From worker 8:	handle_deserialize at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.1/Serialization/src/Serialization.jl:782
      From worker 8:	deserialize at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.1/Serialization/src/Serialization.jl:1221
      From worker 8:	handle_deserialize at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.1/Serialization/src/Serialization.jl:782
      From worker 8:	deserialize at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.1/Serialization/src/Serialization.jl:731
      From worker 8:	#3 at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.1/Serialization/src/Serialization.jl:869
      From worker 8:	ntuple at ./tuple.jl:136
      From worker 8:	deserialize_tuple at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.1/Serialization/src/Serialization.jl:869
      From worker 8:	handle_deserialize at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.1/Serialization/src/Serialization.jl:761
      From worker 8:	deserialize at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.1/Serialization/src/Serialization.jl:731
      From worker 8:	#3 at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.1/Serialization/src/Serialization.jl:869
      From worker 8:	ntuple at ./tuple.jl:136
      From worker 8:	deserialize_tuple at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.1/Serialization/src/Serialization.jl:869
      From worker 8:	handle_deserialize at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.1/Serialization/src/Serialization.jl:761
      From worker 8:	deserialize at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.1/Serialization/src/Serialization.jl:731
      From worker 8:	deserialize_msg at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.1/Distributed/src/messages.jl:99
      From worker 8:	jl_apply at /Users/julia/buildbot/worker/package_macos64/build/src/./julia.h:1571 [inlined]
      From worker 8:	jl_f__apply at /Users/julia/buildbot/worker/package_macos64/build/src/builtins.c:556
      From worker 8:	jl_f__apply_latest at /Users/julia/buildbot/worker/package_macos64/build/src/builtins.c:594
      From worker 8:	#invokelatest#1 at ./essentials.jl:742 [inlined]
      From worker 8:	invokelatest at ./essentials.jl:741 [inlined]
      From worker 8:	message_handler_loop at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.1/Distributed/src/process_messages.jl:160
      From worker 8:	process_tcp_streams at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.1/Distributed/src/process_messages.jl:117
      From worker 8:	#105 at ./task.jl:259
      From worker 8:	jl_fptr_trampoline at /Users/julia/buildbot/worker/package_macos64/build/src/gf.c:1842
      From worker 8:	jl_apply at /Users/julia/buildbot/worker/package_macos64/build/src/./julia.h:1571 [inlined]
      From worker 8:	start_task at /Users/julia/buildbot/worker/package_macos64/build/src/task.c:572
      From worker 8:	Allocations: 313311944 (Pool: 312424770; Big: 887174); GC: 1014
```

**Comment by Keno** [comment]
As always, these kinds of things are super hard to debug without a reproducer. Ideally:
1. See if you can reproduce this on master.
2. If it does, see if you can obtain an rr trace that shows the issue.

**Comment by cwiese** [comment]
I just reproduced this... random.. during loading of IndexedTables on several CPUs ... @async .... so parallel deserializing.   It happens occasionally from same code

**Comment by JeffBezanson** [comment]
Is the stack trace the same as in the original issue here?

**Comment by mmattocks** [comment]
I believe I am experiencing this issue, similar to cwiese in a remotefetch deserialize call to the master process which is @async waiting when it deserializes. It seems weird that the segfault is on the worker. Anyone have any suggestions on how to debug this?

```
      From worker 9:	
      From worker 9:    signal (11): Segmentation fault
      From worker 9:    in expression starting at none:0 
      From worker 9:    gc_mark_loop at /buildworker/worker/package_linux64/build/src/gc.c:2396
      From worker 9:   jl_gc_collect at /buildworker/worker/package_linux64/build/src/gc.c:2899
      From worker 9:    jl_gc_collect at /buildworker/worker/package_linux64/build/src/gc.c:3105
      From worker 9:    maybe_collect at /buildworker/worker/package_linux64/build/src/gc.c:827 [inlined]
      From worker 9:    jl_gc_pool_alloc at /buildworker/worker/package_linux64/build/src/gc.c:1142
      From worker 9:    jl_gc_alloc_ at /buildworker/worker/package_linux64/build/src/julia_internal.h:246 [inlined]
      From worker 9:    jl_gc_alloc at /buildworker/worker/package_linux64/build/src/gc.c:3147
      From worker 9:    jl_alloc_svec_uninit at /buildworker/worker/package_linux64/build/src/simplevector.c:60
      From worker 9:    jl_alloc_svec at /buildworker/worker/package_linux64/build/src/simplevector.c:69
      From worker 9:    save_env at /buildworker/worker/package_linux64/build/src/subtype.c:149
      From worker 9:    forall_exists_subtype at /buildworker/worker/package_linux64/build/src/subtype.c:1447
      From worker 9:    forall_exists_equal at /buildworker/worker/package_linux64/build/src/subtype.c:1392
      From worker 9:    subtype at /buildworker/worker/package_linux64/build/src/subtype.c:1336
      From worker 9:    with_tvar at /buildworker/worker/package_linux64/build/src/subtype.c:702
      From worker 9:⠀ subtype_unionall at /buildworker/worker/package_linux64/build/src/subtype.c:841 [inlined]
      From worker 9:⠀ subtype at /buildworker/worker/package_linux64/build/src/subtype.c:1281
      From worker 9:⠀⠀⠀⠀exists_subtype at /buildworker/worker/package_linux64/build/src/subtype.c:1425 [inlined]
      From worker 9: ⠀⠀⠀forall_exists_subtype at /buildworker/worker/package_linux64/build/src/subtype.c:1453
      From worker 9:   ⠀jl_subtype_env at /buildworker/worker/package_linux64/build/src/subtype.c:1818
      From worker 9:⠀⠀  jl_isa at /buildworker/worker/package_linux64/build/src/subtype.c:2056
      From worker 9:⠀⠀⠀ jl_new_structv at /buildworker/worker/package_linux64/build/src/datatype.c:928
      From worker 9:⠀⠀⠀⠀deserialize at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.4/Serialization/src/Serialization.jl:1315
      From worker 9:⠀⠀⠀⠀_jl_invoke at /buildworker/worker/package_linux64/build/src/gf.c:2145 [inlined]
      From worker 9:⠀⠀⠀⠀jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2323
      From worker 9:⠀⠀⠀⠀handle_deserialize at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.4/Serialization/src/Serialization.jl:799
      From worker 9:⠀⠀⠀⠀deserialize at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.4/Serialization/src/Serialization.jl:735 [inlined]
      From worker 9:⠀⠀⠀⠀deserialize_msg at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.4/Distributed/src/messages.jl:99
      From worker 9:⠀⠀⠀⠀_jl_invoke at /buildworker/worker/package_linux64/build/src/gf.c:2145 [inlined]
      From worker 9:⠀⠀⠀⠀jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2323
      From worker 9:    jl_apply at /buildworker/worker/package_linux64/build/src/julia.h:1700 [inlined]
      From worker 9:    do_apply at /buildworker/worker/package_linux64/build/src/builtins.c:643
      From worker 9:    jl_f__apply_latest at /buildworker/worker/package_linux64/build/src/builtins.c:693
      From worker 9:    invokelatest#1 at ./essentials.jl:712 [inlined] 14.8714
      From worker 9:	invokelatest at ./essentials.jl:711 [inlined]
      From worker 9:    message_handler_loop at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.4/Distributed/src/process_messages.jl:185
      From worker 9:	process_tcp_streams at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.4/Distributed/src/process_messages.jl:142
      From worker 9:    #97 at ./task.jl:358
      From worker 9:    _jl_invoke at /buildworker/worker/package_linux64/build/src/gf.c:2145 [inlined]
      From worker 9:	jl_apply_generic at /buildworker/worker/package_linux64/build/src/gf.c:2323
      From worker 9:	jl_apply at /buildworker/worker/package_linux64/build/src/julia.h:1700 [inlined]
      From worker 9:	start_task at /buildworker/worker/package_linux64/build/src/task.c:687
      From worker 9:	jl_init_root_task at /buildworker/worker/package_linux64/build/src/task.c:1095
      From worker 9:	_julia_init at /buildworker/worker/package_linux64/build/src/init.c:794
      From worker 9:	main at /buildworker/worker/package_linux64/build/ui/repl.c:217
      From worker 9:	__libc_start_main at /lib/x86_64-linux-gnu/libc.so.6 (unknown line)
      From worker 9:	_start at /opt/julia-1.4.2/bin/julia (unknown line)
      From worker 9:	Allocations: 39517687569 (Pool: 39510340308; Big: 7347261); GC: 8019
Worker 9 terminated.
```

**Comment by JeffBezanson** [comment]
Looks like a duplicate of of #35708, fixed in #35737. Pleas reopen if this appears again though.
